### DISCUSSION TITLE
$popup strange behavior

### DISCUSSION THREAD

**Discussion by MasterKia** [uBlockOrigin/uBlock-issues#2270] [General] [answered]
`$popup` strange behavior in uBO

Preferably disable the quick fixes list.

Add:
```adb
||asiatech.ir^$popup,domain=soft98.ir
||hostdl.com^$popup,domain=soft98.ir
```

Go to the footer of `soft98.ir`:
Click on "آسیاتک" => It closes
Click on "هاست دی ال" => It won't close

Both links have `target=_blank`. 

Reproduced using Firefox.

Can anyone reproduce?

**Comment by gorhill** [comment]
Sorry about my previous comment, I read **way** too fast. I can reproduce and I also find this puzzling. Difference with the other link is that there is a redirection from `http://` to `https://`, I need to investigate by stepping in the code.

**Reply by gorhill** [reply]
Ok, it's a long time I went through that popup filtering code, so this is what is happening:

The fact that `hostdl.com` is not filtered as a popup is because the url address of new tab matches the link clicked.

The fact that `asiatech.ir` is filtered as a popup is because the url address of the new tab does not match the link clicked. The link clicked is `asiatech.ir`, while the new tab ends up being `www.asiatech.ir`, so "different" because of the `www` part.

**Reply by MasterKia** [reply]
Edit: reading your comment again, the below seems like a different issue.

The `www.` problem is not limited to $popup, something similar is affecting scriptlet rules as well.

Try `hostdl.com##+js(window-close-if, /^/)` => HostDL popup gets closed
Try `www.hostdl.com##+js(window-close-if, /^/)` => HostDL popup doesn't get closed

One AdGuard maintainer said when they want to limit a rule only to the main domain (i.e. not affect `panel.hostdl.com`), they use `www.`

**Reply by gorhill** [reply]
> Try `www.hostdl.com##+js(window-close-if, /^/)` => HostDL popup doesn't get closed

This is expected? The URL address does not contains `www` part, so there is no match.

**Reply by MasterKia** [reply]
Ok, so one should use `hostdl.com,~panel.hostdl.com,~login.hostdl.com,~stats.hostdl.com,~blog.hostdl.com##+js(...)` then. (hypothetically - rare examples)

**Reply by gorhill** [reply]
You can use this to match either base domain or `www` one:

    hostdl.com##+js(window-close-if, /^https?://(www\.)?hostdl\.com/)

**Reply by gorhill** [reply]
Huh never mind, this does not work. I had to look at the documentation, the scriptlet does not match against the scheme/hostname part, just the path/query part.

**Reply by gorhill** [reply]
I think I could change the behavior to match against the whole URL if the argument is a regex, I perused the filter lists at random and that would not break any of the rare instances of `window-close-if` I found.

**Reply by gorhill** [reply, marked as answer]
https://github.com/gorhill/uBlock/commit/65a056107210796426033ebe2eebb89a98c93a23

**Reply by MasterKia** [reply]
Added to wiki alongside your regex example to match only base domain or www one.
https://github.com/uBlockOrigin/uBlock-issues/wiki/Resources-Library#window-close-ifjs-
